File: Net/SSOExtension/VersionOne.SDK.APIClient.SSOExtension/FormData.cs
using System;
using System.Collections;
using System.Web;

namespace VersionOne.SDK.APIClient.SSOExtension
{
    public abstract class FormData
    {
        private readonly ArrayList _formData = new ArrayList();

        public void Add(string key, string value)
        {
            _formData.Add(string.Format("{0}={1}", key, HttpUtility.UrlEncode(value)));
        }

        public void Add(string key)
        {
            _formData.Add(string.Format("{0}=", key));
        }

        public string ToPostData()
        {
            return String.Join("&", (String[])_formData.ToArray(typeof(string)));
        }

        /// <summary>
        /// Identity Providers have different field names for credentials.   
        /// This abstraction allows the code to make one call to set credentials. 
        /// </summary>
        /// <param name="username"></param>
        /// <param name="password"></param>
        public abstract void SetCredentials(string username, string password);
    }
}

File: Java/VersionOneSSO/src/com/versionone/sso/FormData.java
package com.versionone.sso;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;
import java.util.ArrayList;
import java.util.List;

import org.apache.commons.lang.StringUtils;

public abstract class FormData {

    private List<String> _formData = new ArrayList<String>();
    
    public void add(String key, String value)
    {
        try {
			_formData.add(String.format("%s=%s", key, URLEncoder.encode(value, "UTF-8")));
		} catch (UnsupportedEncodingException e) {
			e.printStackTrace();
			_formData.add(String.format("%s=%s", key, value));
		}
    }
    
    public void add(String key)
    {
        _formData.add(String.format("%s=", key));
    }

    public String getPostData()
    {	
    	return StringUtils.join(_formData, "&");
    }
    
    /**
     * Identity Providers have different field names for credentials.   
     * This abstraction allows the code to make one call to set credentials.  
     * @param username
     * @param password
     */
    public abstract void setCredentials(String username, String password);
}
